blockchain/client.go

```diff
@@ -12,11 +14,39 @@ import (
 	"github.com/ethereum/go-ethereum/rpc"
 )
 
+// Receipt extends go-ethereum/core/types/Receipt struct with BlockNumber
 type Receipt struct {
 	*types.Receipt
 	BlockNumber string
```

Comment:
Looks like a field named `BlockNumber` must have a numeric type. 
This field is using to compare with block confirmation, and both is type-casted into `big.Int`, that is totally meaningless. I'd prefer to represent it as uint64.